### ISSUE TITLE
pyramid module functions upcasts single precision floats to double precision

### ISSUE THREAD

**Issue by rfezzani** [scikit-image/scikit-image#4190]
## Description

Because of the use of `skimage.transform.resize`, the functions of the module `skimage.transform.pyramid` upcast single precision float arrays to double precision.

PR #4128 solves this issue.

## Way to reproduce

```python
>>> from skimage.transform import pyramids
>>> a = np.random.rand(5).astype('float32')
>>> pyramids.pyramid_reduce(a).dtype
dtype('float64')
>>> pyramids.pyramid_expand(a).dtype
dtype('float64')
>>> [img.dtype for img in pyramids.pyramid_gaussian(a)]
[dtype('float32'), dtype('float64'), dtype('float64'), dtype('float64')]
>>> [img.dtype for img in pyramids.pyramid_laplacian(a)]
[dtype('float64'), dtype('float64'), dtype('float64'), dtype('float64')]
```

## Version information

```python
3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]
Linux-4.15.0-64-generic-x86_64-with-debian-buster-sid
scikit-image version: 0.16.dev0
numpy version: 1.16.4
```

**Comment by hmaarrfk** [comment]
Nice finds, and fixed!!!!

**Comment by rfezzani** [comment]
Fixed by #4128
